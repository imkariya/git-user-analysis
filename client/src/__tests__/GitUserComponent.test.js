import React from "react";
import {
  screen,
  render,
  fireEvent,
  act,
  waitForElementToBeRemoved,
} from "../test-utils";
import "@testing-library/jest-dom";
import GitUserComponent from "../component/gitUserComponent";

describe("Take GitUser Component snapshot", () => {
  it("User Detail", async () => {
    render(<GitUserComponent />);
    const inputField = screen.getByRole("textbox");
    const submitBtn = screen.getByTestId("submit-test-id");

    fireEvent.change(inputField, { target: { value: "ankitbihani415" } });
    await act(async () => {
      fireEvent.click(submitBtn);
    });
    await waitForElementToBeRemoved(() =>
      screen.queryAllByTestId("loading-test-id")
    );
    expect(screen.getByTestId("user-profile-test-id")).toMatchSnapshot();
  });
});
