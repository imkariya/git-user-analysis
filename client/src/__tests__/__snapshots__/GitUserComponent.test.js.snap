// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take GitUser Component snapshot User Detail 1`] = `
<div
  class="MuiPaper-root MuiCard-root user-detail-card MuiPaper-outlined MuiPaper-rounded"
  data-testid="user-profile-test-id"
>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <a
      class="image-link"
      href="https://github.com/node"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="node"
        class="avatar"
        height="50px"
        src="https://avatars.githubusercontent.com/u/227278?v=4"
        width="50px"
      />
    </a>
    <div
      class="MuiCardContent-root mb-4 mt-3"
    >
      node
      <div
        class="MuiBox-root MuiBox-root-7 date"
      >
        21/3/2010
      </div>
    </div>
  </div>
  <div
    class="MuiCardContent-root border-top p-2"
  >
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      <div>
        <span>
          Follower: 
          164
        </span>
      </div>
      <div>
        <span>
          Following: 
          334
        </span>
      </div>
      <div>
        <span>
          Total Repo: 
          61
        </span>
      </div>
    </div>
  </div>
</div>
`;
